pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -163,10 +163,15 @@ func scaleResourceArgs(pa *pav1alpha1.PodAutoscaler) (*schema.GroupVersionResour
 }
 
 func (ks *scaler) handleScaleToZero(pa *pav1alpha1.PodAutoscaler, desiredScale int32, config *autoscaler.Config) (int32, bool) {
-	if desiredScale != 0 {
+	if desiredScale > 0 {
 		return desiredScale, true
 	}
 
+	if desiredScale < 0 && !pa.Status.IsActivating() {
+		ks.logger.Info("Metrics are not yet being collected.")
```

Comment:
The level was debug for a reason. Please retain that.
Also this does not belong here. In fact, I'd rather move `if desiredScale > 0` to the call site (I had that in mind for some time).